Check for ride validity in game commands

janisozaur opened the pull request.

IntelOrca commented:
I don't think you can do this, I am not sure if deleting a ride actually shifts all the rides up the array. It might leave a gap.

janisozaur commented:
Yes, that's right, how could I miss that.

Anyway, as the check you mentioned already mentioned was already there I figured out the nature of problem is different and updated accordingly.

`MAX_RIDES` is defined to be 255, but a single byte value can address up to 256 rides, the last one is either uninitialised or points to something else. I added a check against that instead.

Setting a wrong command ride to demolish can cause vanilla code to do something wrong to map animations and can trigger an assert during subsequent `rct2_update`.
